diff --git a/appsearch/compiler/build.gradle b/appsearch/compiler/build.gradle
index b0804004381..43fc3c3d90a 100644
--- a/appsearch/compiler/build.gradle
+++ b/appsearch/compiler/build.gradle
@@ -59,7 +59,7 @@ dependencies {
 androidx {
     name = "AppSearch Compiler"
     type = SoftwareType.ANNOTATION_PROCESSOR
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = '2019'
     description = 'Compiler for classes annotated with @androidx.appsearch.annotation.Document'
     failOnDeprecationWarnings = false
diff --git a/buildSrc/private/src/main/kotlin/androidx/build/testConfiguration/TestSourceSetsHelper.kt b/buildSrc/private/src/main/kotlin/androidx/build/testConfiguration/TestSourceSetsHelper.kt
index 9282773ba8e..df6f26fb014 100644
--- a/buildSrc/private/src/main/kotlin/androidx/build/testConfiguration/TestSourceSetsHelper.kt
+++ b/buildSrc/private/src/main/kotlin/androidx/build/testConfiguration/TestSourceSetsHelper.kt
@@ -33,6 +33,7 @@ internal fun Project.getTestSourceSetsForAndroid(variant: Variant?): List<FileCo
                 testSourceFileCollections.add(files(sourceSet))
             }
             // Add kotlin-android main source set
+            @Suppress("DEPRECATION") // .sourceSets
             extensions
                 .findByType(KotlinAndroidProjectExtension::class.java)
                 ?.sourceSets
@@ -49,6 +50,7 @@ internal fun Project.getTestSourceSetsForAndroid(variant: Variant?): List<FileCo
     }
 
     // Add kotlin-android androidTest source set
+    @Suppress("DEPRECATION") // .sourceSets
     extensions
         .findByType(KotlinAndroidProjectExtension::class.java)
         ?.sourceSets
diff --git a/buildSrc/public/src/main/kotlin/androidx/build/AndroidXConfiguration.kt b/buildSrc/public/src/main/kotlin/androidx/build/AndroidXConfiguration.kt
index 6fece502bef..f1c96195b0b 100644
--- a/buildSrc/public/src/main/kotlin/androidx/build/AndroidXConfiguration.kt
+++ b/buildSrc/public/src/main/kotlin/androidx/build/AndroidXConfiguration.kt
@@ -38,11 +38,11 @@ interface AndroidXConfiguration {
 }
 
 enum class KotlinTarget(val apiVersion: KotlinVersion, val catalogVersion: String) {
-    KOTLIN_2_1(KotlinVersion.KOTLIN_2_1, "kotlin21"),
     KOTLIN_2_2(KotlinVersion.KOTLIN_2_2, "kotlin22"),
     KOTLIN_2_3(KotlinVersion.KOTLIN_2_3, "kotlin23"),
-    DEFAULT(KOTLIN_2_1),
-    LATEST(KOTLIN_2_3);
+    KOTLIN_2_4(KotlinVersion.KOTLIN_2_3, "kotlin24"),
+    DEFAULT(KOTLIN_2_2),
+    LATEST(KOTLIN_2_4);
 
     constructor(
         kotlinTarget: KotlinTarget
diff --git a/buildSrc/shared.gradle b/buildSrc/shared.gradle
index e516bf8575b..bd3400df80f 100644
--- a/buildSrc/shared.gradle
+++ b/buildSrc/shared.gradle
@@ -44,7 +44,7 @@ tasks.withType(KotlinCompile).configureEach { task ->
                 "-Xskip-metadata-version-check",
                 "-Xjdk-release=17",
         )
-        languageVersion.set(KotlinVersion.KOTLIN_2_1)
-        apiVersion.set(KotlinVersion.KOTLIN_2_1)
+        languageVersion.set(KotlinVersion.KOTLIN_2_2)
+        apiVersion.set(KotlinVersion.KOTLIN_2_2)
     }
 }
diff --git a/camera/featurecombinationquery/featurecombinationquery-play-services/build.gradle b/camera/featurecombinationquery/featurecombinationquery-play-services/build.gradle
index b0ccab4a540..f08658bbbb3 100644
--- a/camera/featurecombinationquery/featurecombinationquery-play-services/build.gradle
+++ b/camera/featurecombinationquery/featurecombinationquery-play-services/build.gradle
@@ -51,5 +51,5 @@ androidx {
     inceptionYear = "2024"
     description = "Camera feature combination components for the Camera Library, a " +
             "library providing camera feature combination with Google Play Services dependencies."
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/core/core-location-altitude/build.gradle b/core/core-location-altitude/build.gradle
index 15ebd3cadf8..6e30837dc9a 100644
--- a/core/core-location-altitude/build.gradle
+++ b/core/core-location-altitude/build.gradle
@@ -61,5 +61,5 @@ androidx {
     mavenVersion = LibraryVersions.CORE_LOCATION_ALTITUDE
     inceptionYear = "2022"
     description = "Provides compatibility APIs concerning location altitudes."
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/credentials-play-services-auth/build.gradle b/credentials/credentials-play-services-auth/build.gradle
index 9434793fe1d..5947bce46f1 100644
--- a/credentials/credentials-play-services-auth/build.gradle
+++ b/credentials/credentials-play-services-auth/build.gradle
@@ -67,5 +67,5 @@ androidx {
     inceptionYear = "2022"
     description = "sign into apps using play-services-auth library"
     legacyDisableKotlinStrictApiMode = true
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/credentials/build.gradle b/credentials/credentials/build.gradle
index b00f477ac65..aa4217bd66c 100644
--- a/credentials/credentials/build.gradle
+++ b/credentials/credentials/build.gradle
@@ -61,5 +61,5 @@ androidx {
     description = "Android Credentials Library"
     legacyDisableKotlinStrictApiMode = true
     samples(project(":credentials:credentials-samples"))
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/credentials/samples/build.gradle b/credentials/credentials/samples/build.gradle
index ec75e2dd12a..49f4bfec355 100644
--- a/credentials/credentials/samples/build.gradle
+++ b/credentials/credentials/samples/build.gradle
@@ -49,5 +49,5 @@ androidx {
     type = SoftwareType.SAMPLES
     inceptionYear = "2023"
     description = "Contains the sample code for the Androidx Credentials library"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/providerevents/providerevents-play-services/build.gradle b/credentials/providerevents/providerevents-play-services/build.gradle
index 7378344ff7f..7c4fedcfc13 100644
--- a/credentials/providerevents/providerevents-play-services/build.gradle
+++ b/credentials/providerevents/providerevents-play-services/build.gradle
@@ -53,6 +53,6 @@ androidx {
     name = "androidx.credentials.providerevents:providerevents-play-services"
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2025"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     description = "connect to play-services event updates"
 }
diff --git a/credentials/providerevents/providerevents/build.gradle b/credentials/providerevents/providerevents/build.gradle
index 11da3d55ed2..f7cd6d0e762 100644
--- a/credentials/providerevents/providerevents/build.gradle
+++ b/credentials/providerevents/providerevents/build.gradle
@@ -67,5 +67,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2025"
     description = "extend services that receive updates about credentials"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/registry/registry-digitalcredentials-mdoc/build.gradle b/credentials/registry/registry-digitalcredentials-mdoc/build.gradle
index ad0b571065a..6d893216458 100644
--- a/credentials/registry/registry-digitalcredentials-mdoc/build.gradle
+++ b/credentials/registry/registry-digitalcredentials-mdoc/build.gradle
@@ -49,5 +49,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2024"
     description = "construct the ISO/IEC mdoc format credentials for registry"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/registry/registry-digitalcredentials-openid/build.gradle b/credentials/registry/registry-digitalcredentials-openid/build.gradle
index 1423001f94e..06ec8db90e6 100644
--- a/credentials/registry/registry-digitalcredentials-openid/build.gradle
+++ b/credentials/registry/registry-digitalcredentials-openid/build.gradle
@@ -51,5 +51,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2025"
     description = "register their holder support for OpenID4VP / OpenID4VCI protocols and the corresponding digital credentials with the Credential Manager"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/registry/registry-digitalcredentials-sdjwtvc/build.gradle b/credentials/registry/registry-digitalcredentials-sdjwtvc/build.gradle
index 731f86d7c4f..d2bfc5f6809 100644
--- a/credentials/registry/registry-digitalcredentials-sdjwtvc/build.gradle
+++ b/credentials/registry/registry-digitalcredentials-sdjwtvc/build.gradle
@@ -49,5 +49,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2025"
     description = "construct the IETF SD-JWT VC digital credentials for registry with the Credential Manager"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/registry/registry-provider-play-services/build.gradle b/credentials/registry/registry-provider-play-services/build.gradle
index f27eb291cc4..3a076b91d91 100644
--- a/credentials/registry/registry-provider-play-services/build.gradle
+++ b/credentials/registry/registry-provider-play-services/build.gradle
@@ -55,5 +55,5 @@ androidx {
     inceptionYear = "2024"
     description = "register credentials using play-services-identity-credentials apis"
     legacyDisableKotlinStrictApiMode = true
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/credentials/registry/registry-provider/build.gradle b/credentials/registry/registry-provider/build.gradle
index 4c7284073cf..fa227dbe628 100644
--- a/credentials/registry/registry-provider/build.gradle
+++ b/credentials/registry/registry-provider/build.gradle
@@ -53,5 +53,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2024"
     description = "register digital credentials with CredentialManager to support smooth sign-in, verification, and other user experience"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/navigation3/navigation3-runtime/samples/build.gradle b/navigation3/navigation3-runtime/samples/build.gradle
index 6e32e0e6d88..d0e66239953 100644
--- a/navigation3/navigation3-runtime/samples/build.gradle
+++ b/navigation3/navigation3-runtime/samples/build.gradle
@@ -56,7 +56,7 @@ androidx {
     type = SoftwareType.SAMPLES
     inceptionYear = "2025"
     description = "Contains the sample code for Navigation"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/navigation3/navigation3-ui/samples/build.gradle b/navigation3/navigation3-ui/samples/build.gradle
index f357a40922f..369d7e47ab1 100644
--- a/navigation3/navigation3-ui/samples/build.gradle
+++ b/navigation3/navigation3-ui/samples/build.gradle
@@ -59,7 +59,7 @@ androidx {
     type = SoftwareType.SAMPLES
     inceptionYear = "2025"
     description = "Contains the sample code for Navigation3 UI"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/navigationevent/navigationevent-benchmark/build.gradle b/navigationevent/navigationevent-benchmark/build.gradle
index 682b79c5b3c..a180df4c471 100644
--- a/navigationevent/navigationevent-benchmark/build.gradle
+++ b/navigationevent/navigationevent-benchmark/build.gradle
@@ -54,5 +54,5 @@ android {
 
 androidx {
     type = SoftwareType.BENCHMARK
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/room3/benchmark/build.gradle b/room3/benchmark/build.gradle
index 77e6bb040e1..94a72232912 100644
--- a/room3/benchmark/build.gradle
+++ b/room3/benchmark/build.gradle
@@ -58,7 +58,7 @@ android {
 
 androidx {
     type = SoftwareType.BENCHMARK
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 // TODO(b/430983364): Workaround KSP incremental issues
diff --git a/room3/room3-common/build.gradle b/room3/room3-common/build.gradle
index 6c463c55617..a3d614d8ea9 100644
--- a/room3/room3-common/build.gradle
+++ b/room3/room3-common/build.gradle
@@ -65,5 +65,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "Android Room-Common"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/room3/room3-gradle-plugin/build.gradle b/room3/room3-gradle-plugin/build.gradle
index d417d4a9c7b..549792a90c5 100644
--- a/room3/room3-gradle-plugin/build.gradle
+++ b/room3/room3-gradle-plugin/build.gradle
@@ -97,5 +97,5 @@ androidx {
     minimumGradleVersion = "8.6"
     inceptionYear = "2023"
     description = "Android Room Gradle Plugin"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/room3/room3-gradle-plugin/src/test/java/androidx/room3/gradle/RoomKmpGradlePluginTest.kt b/room3/room3-gradle-plugin/src/test/java/androidx/room3/gradle/RoomKmpGradlePluginTest.kt
index c02d9d12b62..e862d3e3b43 100644
--- a/room3/room3-gradle-plugin/src/test/java/androidx/room3/gradle/RoomKmpGradlePluginTest.kt
+++ b/room3/room3-gradle-plugin/src/test/java/androidx/room3/gradle/RoomKmpGradlePluginTest.kt
@@ -72,7 +72,7 @@ class RoomKmpGradlePluginTest {
             |  }
             |
             |  compilerOptions {
-            |    languageVersion = org.jetbrains.kotlin.gradle.dsl.KotlinVersion.KOTLIN_2_1
+            |    languageVersion = org.jetbrains.kotlin.gradle.dsl.KotlinVersion.KOTLIN_2_2
             |  }
             |}
             |
diff --git a/room3/room3-guava/build.gradle b/room3/room3-guava/build.gradle
index 8e8e54bdf64..4fbea541c6d 100644
--- a/room3/room3-guava/build.gradle
+++ b/room3/room3-guava/build.gradle
@@ -47,7 +47,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2018"
     description = "Android Room Guava"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/room3/room3-livedata/build.gradle b/room3/room3-livedata/build.gradle
index 4b3692db0ac..ca96faa95cd 100644
--- a/room3/room3-livedata/build.gradle
+++ b/room3/room3-livedata/build.gradle
@@ -33,7 +33,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2026"
     description = "Contains the LiveData DAO return type converter for Room."
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/room3/room3-migration/build.gradle b/room3/room3-migration/build.gradle
index 79e89fb832d..baaeaf5a5fc 100644
--- a/room3/room3-migration/build.gradle
+++ b/room3/room3-migration/build.gradle
@@ -72,5 +72,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "Android Room Migration"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/room3/room3-paging-guava/build.gradle b/room3/room3-paging-guava/build.gradle
index b060737c7f4..7dc37554441 100644
--- a/room3/room3-paging-guava/build.gradle
+++ b/room3/room3-paging-guava/build.gradle
@@ -57,7 +57,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2022"
     description = "Guava integration in Room Paging"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/room3/room3-paging-rxjava3/build.gradle b/room3/room3-paging-rxjava3/build.gradle
index 32025e429d4..74d5ff98ca4 100644
--- a/room3/room3-paging-rxjava3/build.gradle
+++ b/room3/room3-paging-rxjava3/build.gradle
@@ -55,7 +55,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2022"
     description = "RxJava3 integration in Room Paging"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/room3/room3-paging/build.gradle b/room3/room3-paging/build.gradle
index bbdfbf87462..93326ab9f22 100644
--- a/room3/room3-paging/build.gradle
+++ b/room3/room3-paging/build.gradle
@@ -89,5 +89,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2021"
     description = "Room Paging integration"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/room3/room3-runtime/build.gradle b/room3/room3-runtime/build.gradle
index 4e32d8eb766..949d7da6b2c 100644
--- a/room3/room3-runtime/build.gradle
+++ b/room3/room3-runtime/build.gradle
@@ -204,7 +204,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "Android Room-Runtime"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 // TODO(b/430983364): Workaround KSP incremental issue
diff --git a/room3/room3-rxjava3/build.gradle b/room3/room3-rxjava3/build.gradle
index 42fd53f71f7..07fa0e8bd2f 100644
--- a/room3/room3-rxjava3/build.gradle
+++ b/room3/room3-rxjava3/build.gradle
@@ -44,7 +44,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2020"
     description = "Android Room RXJava3"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/room3/room3-sqlite-wrapper/build.gradle b/room3/room3-sqlite-wrapper/build.gradle
index 40f6bbf3212..1a1fc0e0b8b 100644
--- a/room3/room3-sqlite-wrapper/build.gradle
+++ b/room3/room3-sqlite-wrapper/build.gradle
@@ -58,7 +58,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2025"
     description = "Use SupportSQLite APIs backed by a RoomDatabase"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 // TODO(b/430983364): Workaround KSP incremental issues
diff --git a/room3/room3-testing/build.gradle b/room3/room3-testing/build.gradle
index 3fdd2c5968a..9563cfe6183 100644
--- a/room3/room3-testing/build.gradle
+++ b/room3/room3-testing/build.gradle
@@ -76,5 +76,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "Android Room Testing"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/savedstate/savedstate-benchmark/build.gradle b/savedstate/savedstate-benchmark/build.gradle
index 925e272b468..47d5b818334 100644
--- a/savedstate/savedstate-benchmark/build.gradle
+++ b/savedstate/savedstate-benchmark/build.gradle
@@ -49,5 +49,5 @@ android {
 
 androidx {
     type = SoftwareType.BENCHMARK
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/sqlite/sqlite-bundled/build.gradle b/sqlite/sqlite-bundled/build.gradle
index f610313f5fa..5e8ed43422a 100644
--- a/sqlite/sqlite-bundled/build.gradle
+++ b/sqlite/sqlite-bundled/build.gradle
@@ -226,5 +226,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2023"
     description = "The implementation of SQLite library using the bundled SQLite."
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/sqlite/sqlite-framework/build.gradle b/sqlite/sqlite-framework/build.gradle
index f4bb70ff06c..f7c36627665 100644
--- a/sqlite/sqlite-framework/build.gradle
+++ b/sqlite/sqlite-framework/build.gradle
@@ -128,7 +128,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "The implementation of SQLite library using the framework code."
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 // b/446711395
diff --git a/sqlite/sqlite-ktx/build.gradle b/sqlite/sqlite-ktx/build.gradle
index 7606f1d6b37..260968e8192 100644
--- a/sqlite/sqlite-ktx/build.gradle
+++ b/sqlite/sqlite-ktx/build.gradle
@@ -42,7 +42,7 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY_ONLY_USED_BY_KOTLIN_CONSUMERS
     inceptionYear = "2018"
     description = "Kotlin extensions for DB"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
 
 android {
diff --git a/sqlite/sqlite/build.gradle b/sqlite/sqlite/build.gradle
index b3a81b4a65f..b4bdf61d972 100644
--- a/sqlite/sqlite/build.gradle
+++ b/sqlite/sqlite/build.gradle
@@ -81,5 +81,5 @@ androidx {
     type = SoftwareType.PUBLISHED_LIBRARY
     inceptionYear = "2017"
     description = "SQLite API"
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
 }
diff --git a/work/work-gcm/build.gradle b/work/work-gcm/build.gradle
index 6b19bac38ce..666d7a9b9da 100644
--- a/work/work-gcm/build.gradle
+++ b/work/work-gcm/build.gradle
@@ -55,7 +55,7 @@ dependencies {
 androidx {
     name = "WorkManager GCM Integration"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2019"
     description = "Android WorkManager GCMNetworkManager Support"
 }
diff --git a/work/work-inspection/build.gradle b/work/work-inspection/build.gradle
index 8c0cc6da565..558e0b14f6f 100644
--- a/work/work-inspection/build.gradle
+++ b/work/work-inspection/build.gradle
@@ -50,7 +50,7 @@ dependencies {
 androidx {
     name = "WorkManager Inspector"
     type = SoftwareType.IDE_PLUGIN
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2020"
     description = "The implementation of WorkManager Inspector."
 }
diff --git a/work/work-multiprocess/build.gradle b/work/work-multiprocess/build.gradle
index 8e24f9d69f0..6541d20cb42 100644
--- a/work/work-multiprocess/build.gradle
+++ b/work/work-multiprocess/build.gradle
@@ -53,7 +53,7 @@ dependencies {
 androidx {
     name = "WorkManager Multiprocess"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2020"
     description = "Android WorkManager runtime library"
     failOnDeprecationWarnings = false
diff --git a/work/work-runtime-ktx/build.gradle b/work/work-runtime-ktx/build.gradle
index b2eb25fed9f..58c05634730 100644
--- a/work/work-runtime-ktx/build.gradle
+++ b/work/work-runtime-ktx/build.gradle
@@ -36,7 +36,7 @@ dependencies {
 androidx {
     name = "WorkManager Kotlin Extensions"
     type = SoftwareType.PUBLISHED_LIBRARY_ONLY_USED_BY_KOTLIN_CONSUMERS
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2018"
     description = "Android WorkManager Kotlin Extensions"
 }
diff --git a/work/work-runtime/build.gradle b/work/work-runtime/build.gradle
index 0cbe0d1ab5f..1dd69671b2a 100644
--- a/work/work-runtime/build.gradle
+++ b/work/work-runtime/build.gradle
@@ -121,7 +121,7 @@ kotlin {
 androidx {
     name = "WorkManager Runtime"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2018"
     description = "Android WorkManager runtime library"
     failOnDeprecationWarnings = false
diff --git a/work/work-rxjava2/build.gradle b/work/work-rxjava2/build.gradle
index e346e8d75df..395c91171d9 100644
--- a/work/work-rxjava2/build.gradle
+++ b/work/work-rxjava2/build.gradle
@@ -42,7 +42,7 @@ dependencies {
 androidx {
     name = "WorkManager RxJava2"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2018"
     description = "Android WorkManager RxJava2 interoperatibility library"
     failOnDeprecationWarnings = false
diff --git a/work/work-rxjava3/build.gradle b/work/work-rxjava3/build.gradle
index 97e335ec8d6..4849dfec559 100644
--- a/work/work-rxjava3/build.gradle
+++ b/work/work-rxjava3/build.gradle
@@ -42,7 +42,7 @@ dependencies {
 androidx {
     name = "WorkManager RxJava3"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2020"
     description = "Android WorkManager RxJava3 interoperatibility library"
 }
diff --git a/work/work-testing/build.gradle b/work/work-testing/build.gradle
index da2a6226f62..1e1eb6126b9 100644
--- a/work/work-testing/build.gradle
+++ b/work/work-testing/build.gradle
@@ -55,7 +55,7 @@ dependencies {
 androidx {
     name = "WorkManager Testing"
     type = SoftwareType.PUBLISHED_LIBRARY
-    kotlinTarget = KotlinTarget.KOTLIN_2_1
+    kotlinTarget = KotlinTarget.KOTLIN_2_2
     inceptionYear = "2018"
     description = "Android WorkManager testing library"
     enableRobolectric()
